projects/sensory_interface/public/app.js

```diff
@@ -53,20 +88,47 @@ function startSoundPlayback() {
     request.send();
 }
 
+// This function is not used currently.
 function playSoundFromData(data) {
-    let source = audioContext.createBufferSource();
+    source = audioContext.createBufferSource();
     audioContext.decodeAudioData(data, function (buffer) {
         source.buffer = buffer;
         source.connect(audioContext.destination);
-        playSoundFromBufferSource(source);
+        playSoundFromBufferSource();
     });
 }
 
-function playSoundFromBufferSource(source) {
+// This function is not used currently.
+function playSoundFromBufferSource() {
     source.start(audioContext.currentTime);
 }
 
-// Currently, this function is not used, we may need it in the future though.
-function stopSoundPlayback(source) {
-    source.stop(audioContext.currentTime);
+function stopSoundPlayback(event) {
+    try {
+        if (oscillator != null) {
+            oscillator.stop(audioContext.currentTime);
+            event.preventDefault();
+        }
+        if (timeOut != null) {
+            window.clearTimeout(timeOut);
+        }
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+function onCellChange(event) {
+    let changedTouch = event.changedTouches[0];
```

Comment:
If we are guaranteed to have at least one element in `changedTouches`, please add it in a comment.
This is to prevent someone else (or, you in the future :) ) considering whether it's a mistake or a case not handled.